src/diffusers/pipelines/flux/pipeline_flux_control.py

```diff
@@ -0,0 +1,891 @@
+                with the following arguments: `callback_on_step_end(self: DiffusionPipeline, step: int, timestep: int,
+                callback_kwargs: Dict)`. `callback_kwargs` will include a list of all tensors as specified by
+                `callback_on_step_end_tensor_inputs`.
+            callback_on_step_end_tensor_inputs (`List`, *optional*):
+                The list of tensor inputs for the `callback_on_step_end` function. The tensors specified in the list
+                will be passed as `callback_kwargs` argument. You will only be able to include variables listed in the
+                `._callback_tensor_inputs` attribute of your pipeline class.
+            max_sequence_length (`int` defaults to 512): Maximum sequence length to use with the `prompt`.
+
+        Examples:
+
+        Returns:
+            [`~pipelines.flux.FluxPipelineOutput`] or `tuple`: [`~pipelines.flux.FluxPipelineOutput`] if `return_dict`
+            is True, otherwise a `tuple`. When returning a tuple, the first element is a list with the generated
+            images.
+        """
+
+        height = height or self.default_sample_size * self.vae_scale_factor
+        width = width or self.default_sample_size * self.vae_scale_factor
+
+        # 1. Check inputs. Raise error if not correct
+        self.check_inputs(
+            prompt,
+            prompt_2,
+            height,
+            width,
+            prompt_embeds=prompt_embeds,
+            pooled_prompt_embeds=pooled_prompt_embeds,
+            callback_on_step_end_tensor_inputs=callback_on_step_end_tensor_inputs,
+            max_sequence_length=max_sequence_length,
+        )
+
+        self._guidance_scale = guidance_scale
+        self._joint_attention_kwargs = joint_attention_kwargs
+        self._interrupt = False
+
+        # 2. Define call parameters
+        if prompt is not None and isinstance(prompt, str):
+            batch_size = 1
+        elif prompt is not None and isinstance(prompt, list):
+            batch_size = len(prompt)
+        else:
+            batch_size = prompt_embeds.shape[0]
+
+        device = self._execution_device
+
+        # 3. Prepare text embeddings
+        lora_scale = (
+            self.joint_attention_kwargs.get("scale", None) if self.joint_attention_kwargs is not None else None
+        )
+        (
+            prompt_embeds,
+            pooled_prompt_embeds,
+            text_ids,
+        ) = self.encode_prompt(
+            prompt=prompt,
+            prompt_2=prompt_2,
+            prompt_embeds=prompt_embeds,
+            pooled_prompt_embeds=pooled_prompt_embeds,
+            device=device,
+            num_images_per_prompt=num_images_per_prompt,
+            max_sequence_length=max_sequence_length,
+            lora_scale=lora_scale,
+        )
+
+        # 4. Prepare latent variables
+        num_channels_latents = self.transformer.config.in_channels // 8
+
+        control_image = self.prepare_image(
```

Comment:
I realize that for flux, pre-generated latents are packed into 3d tensor, so won't be recognized by image processor, we can refactor the image processor in a follow-up PR so that all `image` input can contain the latent form

Reply:
Actually, passing latents to control_image works as expected (I've verified numerically with original implementation as well). It is thanks to line 576 (https://github.com/huggingface/diffusers/pull/9985/files#diff-8f811cfa985866e8aeece00b90679d52566e97767d6ba74f8974be8569b60ee5R576) where we simply return if it's already a torch tensor. Although, that is another bug IMO because you can pass a normal image as tensor and expect the pipeline to work, but since we use `# Copied from` here, we should correct all pipelines together in a separate PR to support latents tensor as well normal image tensor

Reply:
And line 777 takes care of the normal image tensor input case

![image](https://github.com/user-attachments/assets/55d4724e-5982-459f-8e26-f6e88ecd1179)

Reply:
oh I missed that, but it is still wrong (even though it works for latents input) , because we also allow pass the image (4d) as a torch tensor, which needs to be put through image processor for some operations such as normalization etc  - but it is not a big deal, I think most of the time people pass PIL Image. We will need to clean it up across the pipeline though